Compiling issues with deal.ii-master on Frontera

Building ASPECT with deal.ii-master on Frontera gives the following errors.  @tjhei 
(This is not a new issue and can be resolved by using `cmake -DASPECT_PRECOMPILE_HEADERS=OFF ../aspect`, because Frontera's Intel Compiler does not seem to like precompiling the bundled boost library inside deal.II. Therefore, disable precompiling headers. see: https://github.com/geodynamics/aspect/wiki/Installation-on-Frontera)
```
/work2/08254/jiaqi2/frontera/test_candi/install/deal.II-master/include/deal.II/bundled/boost/serialization/throw_exception.hpp(35) (col. 31): internal error: assertion failed: missing typeinfo variable (shared/cfe/edgcpfe/lower_eh.c, line 3077)


compilation aborted for /work2/08254/jiaqi2/frontera/aspect-build/CMakeFiles/aspect.dir/Unity/unity_29_cxx.cxx (code 4)
make[2]: *** [CMakeFiles/aspect.dir/Unity/unity_29_cxx.cxx.o] Error 4
make[2]: *** Waiting for unfinished jobs....
compilation aborted for /work2/08254/jiaqi2/frontera/aspect-build/CMakeFiles/aspect.dir/Unity/unity_28_cxx.cxx (code 4)
make[2]: *** [CMakeFiles/aspect.dir/Unity/unity_28_cxx.cxx.o] Error 4
/work2/08254/jiaqi2/frontera/test_candi/install/deal.II-master/include/deal.II/bundled/boost/serialization/throw_exception.hpp(35) (col. 31): internal error: assertion failed: missing typeinfo variable (shared/cfe/edgcpfe/lower_eh.c, line 3077)


compilation aborted for /work2/08254/jiaqi2/frontera/aspect-build/CMakeFiles/aspect.dir/Unity/unity_26_cxx.cxx (code 4)
make[2]: *** [CMakeFiles/aspect.dir/Unity/unity_26_cxx.cxx.o] Error 4
/work2/08254/jiaqi2/frontera/test_candi/install/deal.II-master/include/deal.II/bundled/boost/serialization/throw_exception.hpp(35) (col. 31): internal error: assertion failed: missing typeinfo variable (shared/cfe/edgcpfe/lower_eh.c, line 3077)


compilation aborted for /work2/08254/jiaqi2/frontera/aspect-build/CMakeFiles/aspect.dir/Unity/unity_25_cxx.cxx (code 4)
make[2]: *** [CMakeFiles/aspect.dir/Unity/unity_25_cxx.cxx.o] Error 4
/work2/08254/jiaqi2/frontera/test_candi/install/deal.II-master/include/deal.II/bundled/boost/serialization/throw_exception.hpp(35) (col. 31): internal error: assertion failed: missing typeinfo variable (shared/cfe/edgcpfe/lower_eh.c, line 3077)


compilation aborted for /work2/08254/jiaqi2/frontera/aspect-build/CMakeFiles/aspect.dir/Unity/unity_27_cxx.cxx (code 4)
make[2]: *** [CMakeFiles/aspect.dir/Unity/unity_27_cxx.cxx.o] Error 4
/work2/08254/jiaqi2/frontera/test_candi/install/deal.II-master/include/deal.II/bundled/boost/serialization/throw_exception.hpp(35) (col. 31): internal error: assertion failed: missing typeinfo variable (shared/cfe/edgcpfe/lower_eh.c, line 3077)


compilation aborted for /work2/08254/jiaqi2/frontera/aspect-build/CMakeFiles/aspect.dir/Unity/unity_24_cxx.cxx (code 4)
make[2]: *** [CMakeFiles/aspect.dir/Unity/unity_24_cxx.cxx.o] Error 4
/work2/08254/jiaqi2/frontera/test_candi/install/deal.II-master/include/deal.II/bundled/boost/serialization/throw_exception.hpp(35) (col. 31): internal error: assertion failed: missing typeinfo variable (shared/cfe/edgcpfe/lower_eh.c, line 3077)


compilation aborted for /work2/08254/jiaqi2/frontera/aspect-build/CMakeFiles/aspect.dir/Unity/unity_40_cxx.cxx (code 4)
make[2]: *** [CMakeFiles/aspect.dir/Unity/unity_40_cxx.cxx.o] Error 4
/work2/08254/jiaqi2/frontera/test_candi/install/deal.II-master/include/deal.II/bundled/boost/serialization/throw_exception.hpp(35) (col. 31): internal error: assertion failed: missing typeinfo variable (shared/cfe/edgcpfe/lower_eh.c, line 3077)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -602,6 +602,12 @@ ELSE()
   SET(ASPECT_UNITY_BUILD OFF CACHE BOOL "Combine source files into less compile targets to speedup compile time. Currently only supported for cmake 3.16 and newer versions." FORCE)
 ENDIF()
 
+IF (ASPECT_PRECOMPILE_HEADERS AND CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+  # Intel 19.1 produces internal compiler errors inside bundled boost with
+  # precompiled headers, so we deactive it:
+  SET(ASPECT_PRECOMPILE_HEADERS OFF CACHE BOOL "" FORCE)
+ENDIF()
+
 IF (ASPECT_PRECOMPILE_HEADERS)
   IF (CMAKE_VERSION VERSION_LESS 3.9)
     MESSAGE(STATUS "Enabling cotire to precompile external header files.")

